ironfish/src/rpc/routes/eth/getTransactionByHash.ts

```diff
@@ -0,0 +1,143 @@
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type GetTransactionByHashRequest = string
+
+export const GetTransactionByHashRequestSchema: yup.StringSchema<GetTransactionByHashRequest> =
+  yup.string().defined()
+
+export type GetTransactionByHashResponse = {
+  blockHash: string
+  blockNumber: string
+  from: string
+  gas: string
+  gasPrice: string
+  maxFeePerGas: string
+  maxPriorityFeePerGas: string
+  hash: string
+  input: string
+  nonce: string
+  to: string | null
+  transactionIndex: string
+  value: string
+  type: string
+  accessList: string[]
+  chainId: string
+  v: string
+  r: string
+  s: string
+  yParity: string
+}
+
+export const GetTransactionByHashResponseSchema: yup.ObjectSchema<GetTransactionByHashResponse> =
+  yup
+    .object({
+      blockHash: yup.string().defined(),
+      blockNumber: yup.string().defined(),
+      from: yup.string().defined(),
+      gas: yup.string().defined(),
+      gasPrice: yup.string().defined(),
+      maxFeePerGas: yup.string().defined(),
+      maxPriorityFeePerGas: yup.string().defined(),
+      hash: yup.string().defined(),
+      input: yup.string().defined(),
+      nonce: yup.string().defined(),
+      to: yup.string().nullable().defined(),
+      transactionIndex: yup.string().defined(),
+      value: yup.string().defined(),
+      type: yup.string().defined(),
+      accessList: yup.array().of(yup.string().defined()).defined(),
+      chainId: yup.string().defined(),
+      v: yup.string().defined(),
+      r: yup.string().defined(),
+      s: yup.string().defined(),
+      yParity: yup.string().defined(),
+    })
+    .defined()
+
+registerEthRoute<typeof GetTransactionByHashRequestSchema, GetTransactionByHashResponse>(
+  `eth_getTransactionByHash`,
+  `${ApiNamespace.eth}/getTransactionByHash`,
+  GetTransactionByHashRequestSchema,
+  async (request, node): Promise<void> => {
+    Assert.isInstanceOf(node, FullNode)
+    const transactionHash =
+      await node.chain.blockchainDb.getEthTransactionHashToTransactionHash(
+        Buffer.from(EthUtils.remove0x(request.data), 'hex'),
+      )
+    if (!transactionHash) {
+      throw new RpcNotFoundError(`Transaction ${request.data} not found`)
+    }
+    const blockHash = await node.chain.getBlockHashByTransactionHash(transactionHash)
+
+    if (!blockHash) {
+      throw new RpcNotFoundError(`Block not found for transaction ${request.data}`)
+    }
+    const blockHeader = await node.chain.getHeader(blockHash)
+    if (!blockHeader) {
+      throw new RpcNotFoundError(`Block header not found for transaction ${request.data}`)
+    }
+    const transactions = await node.chain.getBlockTransactions(blockHeader)
+
+    let transaction:
+      | {
+          transaction: Transaction
+          initialNoteIndex: number
+          sequence: number
+          blockHash: Buffer
+          previousBlockHash: Buffer
+          timestamp: Date
+        }
+      | undefined = undefined
+    let transactionNum = 0
```

Comment:
I think we'll end up creating a util for this because there are multiple eth endpoints that return `transactionIndex`

Reply:
Added